src/ucs/arch/generic/cpu.c

```diff
@@ -11,24 +11,28 @@
 #include <ucs/arch/cpu.h>
 #include <ucs/sys/sys.h>
 #include <ucs/sys/string.h>
+#include <ucs/type/init_once.h>
 
 #define UCS_CPU_CACHE_FILE_FMT     "/sys/devices/system/cpu/cpu%d/cache/index%d/%s"
 #define UCS_CPU_CACHE_LEVEL        "level"
 #define UCS_CPU_CACHE_TYPE         "type"
 #define UCS_CPU_CACHE_SIZE         "size"
 
+#define UCS_CACHE_ENTRY_INIITIALIZER(_level, _type) \
+    {.level = (_level), .type = (_type),            \
+     .once = UCS_INIT_ONCE_INIITIALIZER, .cache = SIZE_MAX, .status = UCS_OK}
 /* cache datatypes */
-typedef struct ucs_cpu_cache_info {
-    int  level;
-    char type[32];
-} ucs_cpu_cache_info_t;
-
-
-static ucs_cpu_cache_info_t ucs_cpu_cache[] = {
-    [UCS_CPU_CACHE_L1d] = {.level = 1, .type = "Data"},
-    [UCS_CPU_CACHE_L1i] = {.level = 1, .type = "Instruction"},
-    [UCS_CPU_CACHE_L2]  = {.level = 2, .type = "Unified"},
-    [UCS_CPU_CACHE_L3]  = {.level = 3, .type = "Unified"}
+struct {
+    int             level;
+    char            type[32];
```

Comment:
Do we really need to save cache name ?

Reply:
so let's make it const char * instead of char[32]?

Reply:
ok, changed to const char*